src/db/db.ts

```diff
@@ -0,0 +1,69 @@
+import { from, Observable, of } from 'rxjs';
+import { Sequelize, ModelCtor } from 'sequelize';
+import { Order, OrderInstance } from './order';
+import { Trade, TradeInstance } from './trade';
+import { Logger } from '../logger';
+import { mergeMap } from 'rxjs/operators';
+
+type InitDBparams = {
+  dataDir?: string;
+  logger: Logger;
+};
+
+type InitDBResponse = {
+  Order: ModelCtor<OrderInstance>;
+  Trade: ModelCtor<TradeInstance>;
+};
+
+const createModels = (sequelize: Sequelize): InitDBResponse => {
+  const models = {
+    Order: Order(sequelize),
+    Trade: Trade(sequelize),
+  };
+
+  models.Order.hasMany(models.Trade, {
+    as: 'orderTrades',
+    foreignKey: 'orderId',
+    constraints: true,
+  });
+  models.Trade.belongsTo(models.Order, {
+    as: 'orderInstance',
+    constraints: true,
+    foreignKey: 'orderId',
+  });
+
+  return models;
+};
+
+let sequelize: Sequelize;
+
+const initDB$ = ({
+  logger,
```

Comment:
Let's change the object argument to individual arguments. This is the pattern I'd like to project to move towards for all functions so that we can provide some of the arguments by currying the function: https://ramdajs.com/docs/#curry

Reply:
As we discussed on the call, we'll do that after a whole refactor.